stock_picking_invoicing/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,499 @@
+}
+
+
+class StockInvoiceOnshipping(models.TransientModel):
+    _name = 'stock.invoice.onshipping'
+    _description = "Stock Invoice Onshipping"
+
+    @api.model
+    def _get_journal_type(self):
+        active_ids = self.env.context.get('active_ids', [])
+        if active_ids:
+            active_ids = active_ids[0]
+        pick_obj = self.env['stock.picking']
+        picking = pick_obj.browse(active_ids)
+        if not picking or not picking.move_lines:
+            return 'sale'
+        pick_type_code = picking.picking_type_id.code
+        line = fields.first(picking.move_lines)
+        if pick_type_code == 'incoming':
+            usage = line.location_id.usage
+        else:
+            usage = line.location_dest_id.usage
+        return JOURNAL_TYPE_MAP.get((pick_type_code, usage), ['sale'])[0]
+
+    journal_type = fields.Selection(
+        selection=[
+            ('purchase_refund', 'Refund Purchase'),
+            ('purchase', 'Create Supplier Invoice'),
+            ('sale_refund', 'Refund Sale'),
+            ('sale', 'Create Customer Invoice')
+        ],
+        default=_get_journal_type,
+        readonly=True,
+    )
+    group = fields.Selection(
+        selection=[
+            ('picking', 'Picking'),
+            ('partner', 'Partner'),
+            ('partner_product', 'Partner/Product'),
+        ],
+        default="picking",
+        help="Group pickings/moves to create invoice(s):\n"
+             "Picking: One invoice per picking;\n"
+             "Partner: One invoice for each picking's partner;\n"
+             "Partner/Product: One invoice per picking's partner and group "
+             "product into a single invoice line.",
+        required=True,
+    )
+    invoice_date = fields.Date()
+    sale_journal = fields.Many2one(
+        comodel_name='account.journal',
+        domain="[('type', '=', 'sale')]",
+        default=lambda self: self._default_journal('sale'),
+        ondelete="cascade",
+    )
+    sale_refund_journal = fields.Many2one(
+        comodel_name='account.journal',
+        domain="[('type', '=', 'sale_refund')]",
+        default=lambda self: self._default_journal('sale_refund'),
+        ondelete="cascade",
+    )
+    purchase_journal = fields.Many2one(
+        comodel_name='account.journal',
+        domain="[('type', '=', 'purchase')]",
+        default=lambda self: self._default_journal('purchase'),
+        ondelete="cascade",
+    )
+    purchase_refund_journal = fields.Many2one(
+        comodel_name='account.journal',
+        domain="[('type', '=', 'purchase_refund')]",
+        default=lambda self: self._default_journal('purchase_refund'),
+        ondelete="cascade",
+    )
+    show_sale_journal = fields.Boolean()
+    show_sale_refund_journal = fields.Boolean(
+        string="Show Refund Sale Journal",
+    )
+    show_purchase_journal = fields.Boolean()
+    show_purchase_refund_journal = fields.Boolean(
+        string="Show Refund Purchase Journal",
+    )
```

Comment:
Only keep 2 journals, 1 for sale and other for purchase, or 1 only and change domain